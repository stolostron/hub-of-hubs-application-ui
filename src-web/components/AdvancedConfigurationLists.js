/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import React from 'react'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import ResourceList from './common/ResourceList'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import getResourceDefinitions from '../definitions'
import { makeGetVisibleTableItemsSelector } from '../reducers/common'
import {
  getSelectedId,
  default as QuerySwitcher
} from './common/QuerySwitcher'
import msgs from '../../nls/platform.properties'
import { withLocale } from '../providers/LocaleProvider'
import { AcmTablePaginationContextProvider } from '@stolostron/ui-components'

const AdvancedConfigurationLists = ({
  secondaryHeaderProps,
  location,
  locale
}) => {
  const defaultOption = 'subscriptions'
  const options = [
    { id: 'subscriptions', resourceType: RESOURCE_TYPES.QUERY_SUBSCRIPTIONS },
    { id: 'placementrules', resourceType: RESOURCE_TYPES.QUERY_PLACEMENTRULES },
    { id: 'channels', resourceType: RESOURCE_TYPES.QUERY_CHANNELS }
  ]

  const selectedId = getSelectedId({ location, options, defaultOption })

  const resourceType = options.find(o => o.id === selectedId).resourceType
  const staticResourceData = getResourceDefinitions(resourceType)
  const getVisibleResources = makeGetVisibleTableItemsSelector(resourceType)

  return (
    <AcmTablePaginationContextProvider localStorageKey="advanced-tables-pagination">
      <ResourceList
        key={selectedId}
        tabs={secondaryHeaderProps.tabs}
        title={secondaryHeaderProps.title}
        resourceType={resourceType}
        staticResourceData={staticResourceData}
        getVisibleResources={getVisibleResources}
      >
        {[
          <QuerySwitcher
            key="switcher"
            options={options.map(({ id }) => ({
              id,
              contents: msgs.get(`resource.${id}`, locale)
            }))}
            defaultOption={defaultOption}
          />
        ]}
      </ResourceList>
    </AcmTablePaginationContextProvider>
  )
}

AdvancedConfigurationLists.propTypes = {
  locale: PropTypes.string,
  location: PropTypes.object,
  secondaryHeaderProps: PropTypes.object
}

export default withLocale(withRouter(AdvancedConfigurationLists))
