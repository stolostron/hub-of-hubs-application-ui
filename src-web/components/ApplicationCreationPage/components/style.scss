/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2020. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
@import 'scss/variables';

.creation-view-controls-labels {
  padding: 0 0.3125rem 1.125rem 0 !important;
}

.shared-resource-warning {
  display: inline-flex;
  color: $color__blue-90;
  line-height: 2rem;

  svg {
    fill: #e6ae3c;
    width: 18px;
    height: 18px;
  }

  p {
    font-size: 14px;
    margin-top: 0.5rem;
  }

  p:first-of-type {
    margin-top: 0rem;
  }

  div:first-of-type {
    margin-right: 5px;
  }

  .warning-font {
    font-weight: 600;
  }

  .code-font {
    font-size: 12px;
    font-family: Menlo, Monaco, "Courier New", monospace;
  }
}

.timeWindow-container {
  font-size: 14px;
  position: relative;

  // Active/Block mode btns
  .timeWindow-mode-container {
    display: inline-grid;
    margin-bottom: 10px;
  }

  .mode-btn {
    margin: 5px 0;
  }

  // Time window config - Days of the week section
  .config-days-section {
    margin-bottom: 20px;
  }

  .config-title-required {
    font-weight: 400;
    color: #c9190b;
    display: inline;
    margin-left: 0.25rem;
  }

  .config-title {
    font-weight: 600;
    font-size: 14px;
    margin-bottom: 10px;
  }

  .time-picker-title {
    display: flex;

    .config-title {
      padding-right: 71px;
    }
  }

  .config-descr {
    font-size: 14px;
    margin-bottom: 10px;
  }

  .config-days-selector {
    width: 50%;
    display: grid;
    grid-template-columns: auto auto;

    .first-col {
      grid-column-start: 1;
    }

    .second-col {
      grid-column-start: 2;
    }
  }

  // Time window config - Timezone section
  .config-timezone-section {
    margin-bottom: 20px;
  }

  .config-timezone-combo-box {
    width: 290px;
    background-color: $acm_background_color;
  }

  // Time window config - Start/End time section
  .config-time-section {
    display: block;
  }

  .config-time-container {
    display: flex;
    margin-bottom: 20px;
  }

  .config-input-time {
    float: left;
    margin-right: 10px;
  }

  .add-time-btn {
    width: 220px;
    height: 40px;
    padding: 5px;
    text-align: center;
    cursor: pointer;
  }

  .btn-disabled {
    cursor: not-allowed !important;
  }

  .add-time-btn-icon {
    margin-left: 7px;
    float: left;
  }

  .add-time-btn-text {
    font-weight: 600;
    font-size: 14px;
    color: #06c;
  }

  .remove-time-btn {
    padding: 12px 5px;
    cursor: pointer;
  }
}
