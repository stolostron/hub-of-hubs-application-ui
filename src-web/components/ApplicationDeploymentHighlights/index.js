/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import React from 'react'

import { connect } from 'react-redux'
import ApplicationDeploymentHighlightsTerminology from './ApplicationDeploymentHighlightsTerminology'

/* eslint-disable react/prop-types */

const mapStateToProps = state => {
  const { QueryApplicationList, secondaryHeader } = state
  return {
    QueryApplicationList,
    secondaryHeader
  }
}

class ApplicationDeploymentHighlights extends React.Component {
  render() {
    return (
      <div id="DeploymentHighlights">
        <ApplicationDeploymentHighlightsTerminology />
      </div>
    )
  }
}

export default connect(mapStateToProps)(ApplicationDeploymentHighlights)
