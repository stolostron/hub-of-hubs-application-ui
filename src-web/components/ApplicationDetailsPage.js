/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import { withRouter } from 'react-router-dom'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import { typedResourcePageDetails } from './common/ResourcePage'

export default withRouter(
  typedResourcePageDetails(RESOURCE_TYPES.QUERY_APPLICATIONS, [], [], [])
)
