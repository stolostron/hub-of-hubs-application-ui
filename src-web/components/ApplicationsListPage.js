/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import React from 'react'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import { typedResourcePageList } from './common/ResourcePage'
import { AcmTablePaginationContextProvider } from '@stolostron/ui-components'
import { PageSection } from '@patternfly/react-core'

const TypedResourcePageList = typedResourcePageList(
  RESOURCE_TYPES.QUERY_APPLICATIONS,
  [],
  [],
  []
)

const ApplicationsListPage = props => (
  <PageSection>
    <AcmTablePaginationContextProvider localStorageKey="application-table-pagination">
      <TypedResourcePageList {...props} />
    </AcmTablePaginationContextProvider>
  </PageSection>
)

export default ApplicationsListPage
