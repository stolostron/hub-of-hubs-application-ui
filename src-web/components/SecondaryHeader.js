/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button } from '@patternfly/react-core'
import {
  AcmPageHeader,
  AcmSecondaryNav,
  AcmSecondaryNavItem
} from '@stolostron/ui-components'
import resources from '../../lib/shared/resources'
import { withRouter, Link } from 'react-router-dom'
import msgs from '../../nls/platform.properties'
import loadable from '@loadable/component'

const AutoRefreshSelect = loadable(() =>
  import(/* webpackChunkName: "autoRefreshSelect" */ './common/AutoRefreshSelect')
)

resources(() => {
  require('../../scss/secondary-header.scss')
})

export class SecondaryHeader extends React.Component {
  constructor(props) {
    super(props)
    this.renderTabs = this.renderTabs.bind(this)
    this.renderLinks = this.renderLinks.bind(this)
  }

  render() {
    const {
      tabs,
      title,
      breadcrumbItems,
      links,
      label,
      labelColor
    } = this.props
    const { locale } = this.context

    const headerArgs = {
      breadcrumb: breadcrumbItems && this.getBreadcrumbs(),
      title: decodeURIComponent(title),
      navigation: tabs &&
        tabs.length > 0 && (
          <AcmSecondaryNav
            aria-label={`${title} ${msgs.get('tabs.label', locale)}`}
          >
            {this.renderTabs()}
          </AcmSecondaryNav>
      ),
      controls: <AutoRefreshSelect route={this.props.location} />,
      actions: (
        <Fragment>
          {links && (
            <div className="secondary-header-links">{this.renderLinks()}</div>
          )}
        </Fragment>
      ),
      switches: (
        <div className="switch-controls">
          <div id="edit-button-portal-id" />
        </div>
      ),
      label,
      labelColor
    }
    return <AcmPageHeader {...headerArgs} />
  }

  getBreadcrumbs() {
    const { breadcrumbItems } = this.props
    return breadcrumbItems
      ? breadcrumbItems.map(breadcrumb => {
        return { text: breadcrumb.label, to: breadcrumb.url }
      })
      : null
  }

  renderLinks() {
    const { links } = this.props,
          { locale } = this.context
    return links.map(link => {
      const {
        id,
        label,
        url,
        kind = 'primary',
        title,
        handleClick = () => this.props.history.push(url)
      } = link
      // if portal, react component will create the button using a portal
      if (kind === 'portal') {
        return !title ? <div key={id} id={id} className="portal" /> : null
      }
      return (
        <Button key={id} id={id} onClick={handleClick} kind={kind}>
          {msgs.get(label, locale)}
        </Button>
      )
    })
  }

  renderTabs() {
    const { tabs, location } = this.props,
          { locale } = this.context
    return tabs.map((tab, idx) => {
      return (
        <AcmSecondaryNavItem
          key={tab.id}
          id={tab.id}
          isActive={(this.getSelectedTab() || 0) === idx}
          onClick={tab.handleClick || undefined}
        >
          <Link
            to={{
              ...location,
              pathname: tab.url,
              state: { tabChange: location }
            }}
            replace={true}
          >
            {msgs.get(tab.label, locale)}
          </Link>
        </AcmSecondaryNavItem>
      )
    })
  }

  getSelectedTab() {
    const { tabs, location } = this.props
    const selectedTab = tabs
      .map((tab, index) => {
        tab.index = index
        return tab
      })
      .filter((tab, index) => {
        if (index === 0) {
          return false
        }
        return location.pathname.startsWith(tab.url)
      })
    return selectedTab[0] && selectedTab[0].index
  }
}

SecondaryHeader.propTypes = {
  breadcrumbItems: PropTypes.array,
  history: PropTypes.object,
  label: PropTypes.string,
  labelColor: PropTypes.string,
  links: PropTypes.array,
  location: PropTypes.object,
  tabs: PropTypes.array,
  title: PropTypes.string,
  tooltip: PropTypes.string
}

SecondaryHeader.contextTypes = {
  locale: PropTypes.string
}

const mapStateToProps = state => {
  return {
    title: state.secondaryHeader.title,
    tabs: state.secondaryHeader.tabs,
    breadcrumbItems: state.secondaryHeader.breadcrumbItems,
    links: state.secondaryHeader.links,
    tooltip: state.secondaryHeader.tooltip,
    role: state.role && state.role.role,
    label: state.secondaryHeader.label,
    labelColor: state.secondaryHeader.labelColor
  }
}

export default withRouter(connect(mapStateToProps)(SecondaryHeader))
