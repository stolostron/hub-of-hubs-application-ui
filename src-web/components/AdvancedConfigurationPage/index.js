/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import React from 'react'
import msgs from '../../../nls/platform.properties'
import { connect } from 'react-redux'
import resources from '../../../lib/shared/resources'
import {
  updateSecondaryHeader,
  clearSuccessFinished
} from '../../actions/common'

import ApplicationDeploymentHighlights from '../ApplicationDeploymentHighlights'
import { withRouter } from 'react-router-dom'
import AdvancedConfigurationLists from '../AdvancedConfigurationLists'
import { PageSection, Stack, StackItem } from '@patternfly/react-core'

/* eslint-disable react/prop-types */

resources(() => {
  require('./style.scss')
})

const mapDispatchToProps = dispatch => {
  return {
    clearSuccessFinished: () => clearSuccessFinished(dispatch),
    updateSecondaryHeaderFn: (title, tabs) =>
      dispatch(updateSecondaryHeader(title, tabs, null, null, null, null))
  }
}

class AdvancedConfigurationPage extends React.Component {
  componentDidMount() {
    const {
      secondaryHeaderProps,
      updateSecondaryHeaderFn,
      locale
    } = this.props

    updateSecondaryHeaderFn(
      msgs.get(secondaryHeaderProps.title, locale),
      secondaryHeaderProps.tabs
    )
  }

  componentWillUnmount() {
    this.mutateFinished()
  }

  mutateFinished() {
    this.props.clearSuccessFinished()
  }

  render() {
    const { secondaryHeaderProps } = this.props

    return (
      <PageSection>
        <Stack hasGutter>
          <StackItem>
            <ApplicationDeploymentHighlights />
          </StackItem>
          <StackItem>
            <AdvancedConfigurationLists
              secondaryHeaderProps={secondaryHeaderProps}
            />
          </StackItem>
        </Stack>
      </PageSection>
    )
  }
}

export default withRouter(
  connect(null, mapDispatchToProps)(AdvancedConfigurationPage)
)
