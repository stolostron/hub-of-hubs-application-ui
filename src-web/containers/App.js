/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

// seems to be an issue with this rule and redux
/* eslint-disable import/no-named-as-default */

import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { AcmHeader, AcmPage, AcmRoute } from '@stolostron/ui-components'
import queryString from 'query-string'
import SecondaryHeader from '../components/SecondaryHeader'
import { Route, Switch, Redirect, withRouter } from 'react-router-dom'
import { withLocale } from '../providers/LocaleProvider'
import resources from '../../lib/shared/resources'
import client from '../../lib/shared/client'
import loadable from '@loadable/component'
import config from '../../lib/shared/config'
import _ from 'lodash'

export const ActionModalApollo = loadable(() =>
  import(/* webpackChunkName: "actionModalApollo" */ '../components/common-apollo/ActionModalApollo')
)

export const ApplicationsListPage = loadable(() =>
  import(/* webpackChunkName: "applicationsListPage" */ '../components/ApplicationsListPage')
)

export const ApplicationDetailsPage = loadable(() =>
  import(/* webpackChunkName: "applicationDetailsPage" */ '../components/ApplicationDetailsPage')
)

export const ApplicationCreationPage = loadable(() =>
  import(/* webpackChunkName: "applicationCreatePage" */ '../components/ApplicationCreationPage/ApplicationCreationPage')
)

export const AdvancedConfigurationPage = loadable(() =>
  import(/* webpackChunkName: "advancedConfigurationPage" */ '../components/AdvancedConfigurationPage')
)

export const ArgoCreationPage = loadable(() =>
  import(/* webpackChunkName: "ArgoCreationPage" */ '../components/ArgoCreationPage/ArgoCreationPage')
)

resources(() => {
  require('../../scss/common.scss')
})

class App extends React.Component {
  constructor(props) {
    super(props)
    if (client) {
      try {
        this.serverProps = JSON.parse(
          document.getElementById('propshcm').textContent
        )
      } catch (e) {
        this.serverProps = null
      }
    }
  }

  getChildContext() {
    return {
      locale: this.getServerProps().context.locale
    }
  }

  getServerProps() {
    if (client && this.serverProps) {
      return this.serverProps
    }
    if (this.props.serverProps) {
      return this.props.serverProps
    }
    return this.props.staticContext
  }

  render() {
    const serverProps = this.getServerProps()
    const { locale, location, match } = this.props

    const BASE_PAGE_PATH = match.url.replace(/\/$/, '')
    const allApplicationsTabs = [
      {
        id: 'overview',
        label: 'description.title.overview',
        url: BASE_PAGE_PATH
      },
      {
        id: 'advanced',
        label: 'description.title.advancedConfiguration',
        url: `${BASE_PAGE_PATH}/advanced`
      }
    ]

    const getSingleApplicationBasePath = params => {
      return `${BASE_PAGE_PATH}/${params.namespace}/${params.name}`
    }

    const getSingleApplicationTabs = params => {
      const searchParams = queryString.parse(_.get(location, 'search'))
      const { apiVersion, applicationset, cluster } = searchParams
      const SINGLE_APP_BASE_PAGE_PATH = getSingleApplicationBasePath(params)
      const overviewTab = [
        {
          id: 'overview',
          label: 'description.title.overview',
          url: SINGLE_APP_BASE_PAGE_PATH
        }
      ]
      if (apiVersion === 'argoproj.io/v1alpha1') {
        if (!applicationset || cluster) {
          return overviewTab
        }
      }
      return _.concat(overviewTab, {
        id: 'editor',
        label: 'description.title.editor',
        url: `${SINGLE_APP_BASE_PAGE_PATH}/edit`
      })
    }

    const applicationsTitle = 'routes.applications'

    return (
      <AcmPage header={<SecondaryHeader />}>
        <Switch>
          <Route
            exact
            path={`${BASE_PAGE_PATH}`}
            render={params => (
              <ApplicationsListPage
                params={params}
                serverProps={this.getServerProps()}
                secondaryHeaderProps={{
                  title: applicationsTitle,
                  tabs: allApplicationsTabs
                }}
              />
            )}
          />
          <Route
            exact
            path={`${BASE_PAGE_PATH}/advanced`}
            render={params => (
              <AdvancedConfigurationPage
                params={params}
                serverProps={serverProps}
                secondaryHeaderProps={{
                  title: applicationsTitle,
                  tabs: allApplicationsTabs
                }}
                locale={locale}
              />
            )}
          />
          <Route
            exact
            path={`${BASE_PAGE_PATH}/create`}
            render={params => (
              <ApplicationCreationPage
                params={params}
                serverProps={this.getServerProps()}
                secondaryHeaderProps={{ title: 'application.create.title' }}
              />
            )}
          />
          <Route
            exact
            path={`${BASE_PAGE_PATH}/argoappset`}
            render={params => (
              <ArgoCreationPage
                params={params}
                serverProps={this.getServerProps()}
                secondaryHeaderProps={{
                  title: 'argo.create.title',
                  label: 'creation.app.section.techPreview',
                  labelColor: 'creation.app.section.techPreview.color'
                }}
              />
            )}
          />
          <Route
            exact
            path={`${BASE_PAGE_PATH}/:namespace/:name`}
            render={params => (
              <ApplicationDetailsPage
                params={params}
                serverProps={this.getServerProps()}
                secondaryHeaderProps={{
                  title: applicationsTitle,
                  tabs: getSingleApplicationTabs(params.match.params)
                }}
              />
            )}
          />
          <Route
            exact
            path={`${BASE_PAGE_PATH}/:namespace/:name/edit`}
            render={routeProps => {
              const params = queryString.parse(routeProps.location.search)
              return params.apiVersion === 'argoproj.io/v1alpha1' ? (
                <ArgoCreationPage
                  params={params}
                  serverProps={this.getServerProps()}
                  secondaryHeaderProps={{
                    title: 'argo.create.title',
                    tabs: getSingleApplicationTabs(routeProps.match.params)
                  }}
                />
              ) : (
                <ApplicationCreationPage
                  params={routeProps}
                  serverProps={this.getServerProps()}
                  secondaryHeaderProps={{
                    title: 'application.create.title',
                    tabs: getSingleApplicationTabs(routeProps.match.params)
                  }}
                />
              )
            }}
          />
          <Redirect to={`${config.contextPath}/welcome`} />
        </Switch>
        <ActionModalApollo locale={serverProps.context.locale} />
      </AcmPage>
    )
  }
}

App.propTypes = {
  locale: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object,
  serverProps: PropTypes.object,
  staticContext: PropTypes.object
}

App.childContextTypes = {
  locale: PropTypes.string
}

const mapStateToProps = state => {
  return {
    user: state.user
  }
}

const getAcmRoute = props => {
  let path = ''
  if (client) {
    path = window.location.pathname
  } else {
    path = props.url
  }
  if (path.includes(config.contextPath)) {
    return AcmRoute.Applications
  }
  return AcmRoute.Welcome
}

const Container = Component =>
  withRouter(withLocale(connect(mapStateToProps)(Component)))
const AppContainer = Container(App)

const AppComponent = props => (
  <AcmHeader route={getAcmRoute(props)}>
    <Route
      path={config.contextPath}
      serverProps={props}
      component={AppContainer}
    />
  </AcmHeader>
)
AppComponent.displayName = 'AppComponent'
export default AppComponent
