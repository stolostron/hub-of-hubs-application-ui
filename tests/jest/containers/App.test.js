/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2016, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
const React = require("../../../node_modules/react");

import App from "../../../src-web/containers/App";

import { shallow } from "enzyme";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import toJson from "enzyme-to-json";

import configureMockStore from "redux-mock-store";

import { reduxStoreAppPipeline, serverProps } from "../components/TestingData";

const mockStore = configureMockStore();
const storeApp = mockStore(reduxStoreAppPipeline);

const props = {
  className: "hcmapplication",
  content: "test",
  serverProps: serverProps
};

describe("App", () => {
  it("App renders correctly with data on single app.", () => {
    const match = {
      path: "/multicloud",
      url: "/multicloud",
      isExact: false,
      params: {}
    };
    const staticContext = {
      title: "Multicloud Manager",
      context: {
        locale: "en"
      },
      xsrfToken: "abc"
    };

    const component = shallow(
      <Provider store={storeApp}>
        <BrowserRouter>
          <App match={match} staticContext={staticContext} {...props} />
        </BrowserRouter>
      </Provider>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
