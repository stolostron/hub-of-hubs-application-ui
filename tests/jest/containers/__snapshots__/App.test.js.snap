// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders correctly with data on single app. 1`] = `
<BrowserRouter>
  <AppComponent
    className="hcmapplication"
    content="test"
    match={
      Object {
        "isExact": false,
        "params": Object {},
        "path": "/multicloud",
        "url": "/multicloud",
      }
    }
    serverProps={
      Object {
        "context": Object {
          "locale": "en-US",
        },
        "xsrfToken": "test",
      }
    }
    staticContext={
      Object {
        "context": Object {
          "locale": "en",
        },
        "title": "Multicloud Manager",
        "xsrfToken": "abc",
      }
    }
  />
</BrowserRouter>
`;
